runtime/src/message_processor.rs

```diff
@@ -56,7 +63,7 @@ impl MessageProcessor {
         sysvars: &[(Pubkey, Vec<u8>)],
         blockhash: Hash,
         lamports_per_signature: u64,
-    ) -> Result<(), TransactionError> {
+    ) -> Result<ProcessedMessageInfo, TransactionError> {
```

Comment:
As already discussed on Slack, this part here will most likely change when account are passed back directly (not by `Rc` through the parameters). Because the account need to be always returned and this struct only on success, it would become somewhat messy. But I am not sure yet, what a good solution would be.

Reply:
Remind why we are passing the accounts back in the return value rather then shared access with `Rc`?

Reply:
`Rc` around `accounts` won't work anymore in ABIv2 (see #21706 and #21927).

Reply:
I didn't see an explanation in those prs as to why we must get rid of `Rc`

Reply:
You are right, I removed that when splitting the PRs, will add it back in.
The short version is that we can't serialize a `Rc`, so we would have to wrap / unwrap it twice in `process_message()` which is completely unnecessary.